tests/routes/test_vespasearch.py

```diff
@@ -0,0 +1,743 @@
+) -> Mapping[str, Sequence[str]]:
+    meta_value = {}
+    for k in taxonomy:
+        allowed_values = taxonomy[k]["allowed_values"]
+        element_count = random.randint(0, len(allowed_values))
+        meta_value[k] = random.sample(allowed_values, element_count)
+    return meta_value
+
+
+def _create_family_metadata(db, family):
+    if family["fields"]["family_source"] == "UNFCCC":
+        taxonomy = get_unf3c_taxonomy()
+    elif family["fields"]["family_source"] == "CCLW":
+        taxonomy = get_cclw_taxonomy()
+    else:
+        raise ValueError(
+            f"Could not get taxonomy for: {family['fields']['family_source']}"
+        )
+    metadata_value = _generate_synthetic_metadata(taxonomy)
+
+    family_import_id = family["fields"]["family_import_id"]
+    family_metadata = FamilyMetadata(
+        family_import_id=family_import_id,
+        taxonomy_id=1,
+        value=metadata_value,
+    )
+    db.add(family_metadata)
+    db.commit()
+
+
+def _create_document(db, doc, family):
+    physical_document = PhysicalDocument(
+        title="doc name",
+        cdn_object="cdn_object",
+        md5_sum="md5_sum",
+        source_url="source_url",
+        content_type="content_type",
+    )
+
+    db.add(physical_document)
+    db.commit()
+    db.refresh(physical_document)
+    physical_document_language = PhysicalDocumentLanguage(
+        language_id=1826,  # English
+        document_id=physical_document.id,
+        source=LanguageSource.USER,
+        visible=True,
+    )
+    db.add(physical_document_language)
+    db.commit()
+    db.refresh(physical_document_language)
+    db.refresh(physical_document)
+
+    if len(family["fields"]["document_languages"]) > 0:
+        variant = Variant(variant_name="Official Translation", description="")
+    else:
+        variant = Variant(variant_name="Original Language", description="")
+    db.merge(variant)
+    db.commit()
+
+    doc_type = FamilyDocumentType(
+        name=family["fields"]["family_category"], description=""
+    )
+    db.merge(doc_type)
+    db.commit()
+
+    family_import_id = family["fields"]["family_import_id"]
+    doc_import_id = family["fields"]["document_import_id"]
+
+    family_document = FamilyDocument(
+        family_import_id=family_import_id,
+        physical_document_id=physical_document.id,
+        import_id=doc_import_id,
+        variant_name=variant.variant_name,
+        document_status=DocumentStatus.PUBLISHED,
+        document_type=doc_type.name,
+    )
+
+    family_document_slug = Slug(
+        name=f"fd_{_parse_id(doc)}",
+        family_import_id=None,
+        family_document_import_id=doc_import_id,
+    )
+
+    db.add(family_document)
+    db.commit()
+    db.add(family_document_slug)
+    db.commit()
+    db.refresh(family_document)
+
+
+def _make_search_request(client, params):
+    response = client.post(SEARCH_ENDPOINT, json=params)
+    assert response.status_code == 200
+    return response.json()
```

Comment:
Above this point in the module there's a lot of setting up rds and util functions. Thoughts on splitting out and making it very clear what's happening?

Reply:
Yep, nice idea